handbook/engineering/distribution/goals.md

```diff
@@ -4,11 +4,21 @@ Goals are continuously updated and reviewed. If you find these goals do not refl
 
 ## Medium-term goals (~3-6 Months)
 
+### [Improve deployment pipeline UX](https://github.com/orgs/sourcegraph/projects/96)
+
+- **Owner**: TBD
+- **Status**: Not started. Unknown amount of work.
+- **Outcomes**:
+  - Its simple to identify if a commit has been deployed to an environment
+  - We can trigger a deployment via a `/` command in Slack
```

Comment:
I was under the impression that the `/`-command thing was primarily for fast rollbacks, which I think is a nice idea - I think `/`-command just for deployment seems like a crutch for what should be a more transparent deployment process (what we should aim for, ie the point above, which should probably encapsulate "how my commit gets deployed is obvious")

Reply:
It depends on how you look at it, I think a simpler https://github.blog/2015-06-02-deploying-branches-to-github-com/ is highly desirable and gives a lot of visibility onto what is deploying, when it finishes, etc.

Reply:
I think we can the visibility on deployment while keeping the automated fashion of "what's in master is what is out there" (which I think is a lot less confusing than the model in the blog post, which is "any branch could be out there" - I'm also curious about whether they still use that model, given the post is 5 years old and some of the repos linked are now archived 🤔 )

Reply:
I dont know if they do, and I would not map it 1 to 1, I think a simplified version of that works quite well. Its similar to what was shared here https://technology.riotgames.com/news/how-riot-games-uses-slack

Reply:
👍 from me on keeping it automatic. 

I think we should strive to have anything committed to `main` be ready for production and therefore can be deployed to any environment. We can use feature flags to actually enable / disable it per environment. For example if we want to demo a new feature on demo.sourcegraph.com before it's public.

Reply:
I'm a bit torn between whether we actually need a `/`command. If deployment is automatic then I don't think we need it for that. Fast rollbacks are nice, but it means that we are out of sync with what is committed vs what is deployed. If CI was fast enough (I know that's a stretch) then a rollback could be done via a git revert.

Reply:
There are 2 types of rollback from my point of view.

**Roll forward**: Revert the code, but you are actually just creating a new artifact that removes a feature/change
**Roll back**: Redeploy an existing artifact

Reply:
In that case in an ideal world I would prefer Roll Forward 😄

Reply:
Roll back seems like the same thing, but is "instant" vs needing any artifacts to be built. I don't think roll forward is achievable.

Note: our infrastructure is in a repo. So we will always be tracking the `release` branch of that. So for example a roll back would a roll forward in the infra repo :P

Reply:
A roll back could also just be a direct git revert without a PR?

Reply:
IMO its still a "roll forward" as you have to rebuild the artifacts, you are creating a new version that lacks a feature, not restoring an older version.